includes/integrations/wc-blocks/class-orddd-lite-delivery-blocks.php

```diff
@@ -0,0 +1,337 @@
+<?php
+/**
+ * Order Delivery Date for WooCommerce
+ *
+ * Processes performed on the frontend checkout page.
+ *
+ * @author      Tyche Softwares
+ * @package     Order-Delivery-Date-Pro-for-WooCommerce/Frontend/Checkout-Page-Processes
+ * @since       1.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Order Delivery date in WooCommerce checkout block.
+ */
+class ORDDD_Lite_Delivery_Blocks {
+
+	/**
+	 * Constructor function
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_store_api_checkout_update_order_from_request', array( &$this, 'orddd_lite_update_block_order_meta_delivery_date' ), 10, 2 );
+		add_action( 'woocommerce_store_api_checkout_update_order_from_request', array( &$this, 'orddd_lite_orddd_update_order_meta_time_slot' ), 10, 2 );
+		add_action( 'woocommerce_store_api_checkout_update_order_from_request', array( &$this, 'orddd_lite_validate_date_wpefield_cart_block' ), 10, 2 );
+	}
+
+	/**
+	 * Save the delivery date & time slot to order meta in WC Checkout block.
+	 *
+	 * @param object $order Order object.
+	 * @param array  $request Order data to be saved.
+	 * @return void
+	 */
+	public static function orddd_lite_update_block_order_meta_delivery_date( $order, $request ) {
+		$order_id = $order->get_id();
+		if ( isset( $request['extensions']['order-delivery-date'] ) && '' !== $request['extensions']['order-delivery-date'] ) { //phpcs:ignore
+			$delivery_date = '';
+			$date_format   = 'dd-mm-y';
+			$data          = isset( $request['extensions']['order-delivery-date'] ) ? $request['extensions']['order-delivery-date'] : array();
+
+			if ( isset( $data['h_deliverydate'] ) ) { //phpcs:ignore
+				$delivery_date = sanitize_text_field( wp_unslash( $data['h_deliverydate'] ) ); //phpcs:ignore
+			}
+
+			Orddd_Lite_Common::update_order_meta( $order_id, get_option( 'orddd_lite_delivery_date_field_label' ), sanitize_text_field( wp_unslash( $data['e_deliverydate'] ) ), $order ); //phpcs:ignore
+
+			$timestamp = Orddd_Lite_Common::orddd_lite_get_timestamp( $delivery_date, $date_format );
+			Orddd_Lite_Common::update_order_meta( $order_id, '_orddd_lite_timestamp', $timestamp, $order );
+			self::orddd_lite_update_lockout_days( $delivery_date );
+		} else {
+			$delivery_enabled    = Orddd_Lite_Common::orddd_lite_is_delivery_enabled();
+			$is_delivery_enabled = 'yes';
+			if ( 'no' === $delivery_enabled ) {
+				$is_delivery_enabled = 'no';
+			}
+
+			if ( 'yes' === $is_delivery_enabled ) {
+				Orddd_Lite_Common::update_order_meta( $order_id, get_option( 'orddd_delivery_date_field_label' ), '', $order );
+			}
+		}
+		$order->save();
+	}
+
+	/**
+	 * Updates the lockout for the delivery date in the options table
+	 *
+	 * @param string $delivery_date Selected Delivery Date.
+	 * @since 1.5
+	 */
+	public static function orddd_lite_update_lockout_days( $delivery_date ) {
```

Comment:
This is already defined in class-orddd-lite-process.php. Please call the same function instead of creating it again.